### PR TITLE
ci: add pyodide build

### PR DESCRIPTION
**Pull request by henryiii**
Testing this.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -5,6 +5,9 @@ on:
     branches:
       - main
   workflow_dispatch:
+  pull_request:
+    paths:
+      - .github/workflows/release.yml
```
Comment (HDembinski) [inline .github/workflows/release.yml:10]: Nice

Comment (henryiii) [reply .github/workflows/release.yml:10]: To give credit, I first saw this here: https://github.com/astral-sh/uv/blob/fbf562d5a86a58d857ed31545f3faed9ed837da7/.github/workflows/build-binaries.yml#L16-L20 - it's great for working on this file, saves so much time going to manually trigger workflows. I think having it trigger on pyproject.toml would be a bit much, but this is an easy win.

Comment (henryiii) [reply .github/workflows/release.yml:10]: I think everything is fixed upstream, let's see! :)

**Comment by henryiii** [comment]
```
  ______________________ test_NormalConstraint_bad_input_4 _______________________
      def test_NormalConstraint_bad_input_4():
  >       with pytest.raises(ValueError, match="positive definite"):
  E       Failed: DID NOT RAISE <class 'ValueError'>
  /home/runner/work/iminuit/iminuit/tests/test_cost.py:1573: Failed
  ______________________ test_NormalConstraint_bad_input_5 _______________________
      def test_NormalConstraint_bad_input_5():
          n = NormalConstraint(["a", "b"], [1, 2], [[1, 0], [0, 1]])
      
  >       with pytest.raises(ValueError, match="positive definite"):
  E       Failed: DID NOT RAISE <class 'ValueError'>
  /home/runner/work/iminuit/iminuit/tests/test_cost.py:1580: Failed
  ____________________________ test_positive_definite ____________________________
      def test_positive_definite():
          assert util.is_positive_definite([[1, 0], [0, 1]])
  >       assert not util.is_positive_definite([[1, 1], [1, 1]])
  E       assert not True
  E        +  where True = <function is_positive_definite at 0x2898118>([[1, 1], [1, 1]])
  E        +    where <function is_positive_definite at 0x2898118> = util.is_positive_definite
  /home/runner/work/iminuit/iminuit/tests/test_util.py:776: AssertionError
```

**Comment by HDembinski** [comment]
> This is the only failing part, don't know if it's directly pyodide's fault:
> 
> ```
>   ______________________ test_NormalConstraint_bad_input_4 _______________________
>       def test_NormalConstraint_bad_input_4():
>   >       with pytest.raises(ValueError, match="positive definite"):
>   E       Failed: DID NOT RAISE <class 'ValueError'>
>   /home/runner/work/iminuit/iminuit/tests/test_cost.py:1573: Failed
>   ______________________ test_NormalConstraint_bad_input_5 _______________________
>       def test_NormalConstraint_bad_input_5():
>           n = NormalConstraint(["a", "b"], [1, 2], [[1, 0], [0, 1]])
>       
>   >       with pytest.raises(ValueError, match="positive definite"):
>   E       Failed: DID NOT RAISE <class 'ValueError'>
>   /home/runner/work/iminuit/iminuit/tests/test_cost.py:1580: Failed
>   ____________________________ test_positive_definite ____________________________
>       def test_positive_definite():
>           assert util.is_positive_definite([[1, 0], [0, 1]])
>   >       assert not util.is_positive_definite([[1, 1], [1, 1]])
>   E       assert not True
>   E        +  where True = <function is_positive_definite at 0x2898118>([[1, 1], [1, 1]])
>   E        +    where <function is_positive_definite at 0x2898118> = util.is_positive_definite
>   /home/runner/work/iminuit/iminuit/tests/test_util.py:776: AssertionError
> ```

The two errors seem related to `util.is_positive_definite`, which is implemented as follows:
```py
    m = np.atleast_2d(m)
    if np.all(m.T == m):
        # maybe check this first https://en.wikipedia.org/wiki/Diagonally_dominant_matrix
        # and only try cholesky if that fails
        try:
            np.linalg.cholesky(m)
        except np.linalg.LinAlgError:
            return False
        return True
    return False
```
This returns true in those failing cases although it should return false. Apparently, `np.linalg.cholesky(m)` is not raising `LinAlgError` on Pyiodide.

**Comment by HDembinski** [comment]
Perhaps the line `np.linalg.cholesky(m)` is non-blocking in Pyiodide. This code assumes it is blocking, like any other normal Python statement.

**Comment by henryiii** [comment]
I bet NumPy is compiled with exceptions turned off; maybe this function actually raises the exception from the compiled code?

**Comment by HDembinski** [comment]
@henryiii I made a patch which may work around the failing test in pyodide, but I cannot test it because the pyodide build now fails because of some build-system issue. I am closing this PR now, you can reopen it when you find the time to finish this.

**Pull request closed without merging**